### ISSUE TITLE
It says it merges ... but the result is only part 1

### ISSUE THREAD

**Issue by drandreaskrueger** [mbunkus/mkvtoolnix#1329]
MKVToolNix GUI 8.1.0 says it merges ... but the result is only part 1

I am trying to merge two .mts files which were written by a Sat-TV-receiver (XORO HRS 8560):
part1.mts 4,194,048 KB (01:58:00 hours)
part2.mts 2,168,037 KB (01:02:00 hours)

after ~7 minutes of mkvmerge v8.1.0 results in:
result.mkv 5,461,545 KB (01:58:00 hours)

as the result is bigger, I would expect it to really be the concatenation of the aforement files; but still strangely it is shorter than the simple sum of KB. And the main problem is not that, but:

The resulting file plays as if it is only part1,
i.e. the timeline has identical length (01:58:00), and it stops playing at 01:58:00.

I would have expected the result.mkv to be 03:00:00 hours long, that was why I used mkvmerge in the first place *lol*

I have already tried (rather random) clicking of options, e.g.
* NALU size length force 4 bytes
* Indexing (cues): for all frames

but after (now already 4 times) 7 minutes ... the outcome is always the same.

Below is the log output.

Thanks a million! This looks like a really great program, please help me to make it work for my use case. Thanks.

Have a great summer!

Andreas

P.S.: My final goal is to have 1 small file per TV recording; so concatenate 2-7 huge raw files & compress to approx 1GB videofiles. If there is an option / software to do all that in one go, I'd be most happy to learn about that. Thanks!

----

--- Output of job 'merging to file »myOutput.mkv« in directory »C:/«' started on 2015-07-01 14:56:48 ---
mkvmerge v8.1.0 ('Psychedelic Postcard') 64bit
'E:/part1.mts': Using the demultiplexer for the format 'MPEG transport stream'.
'E:/part2.mts': Using the demultiplexer for the format 'MPEG transport stream'.
'E:/part1.mts' track 12: Using the output module for the format 'MPEG-1/2'.
'E:/part1.mts' track 13: Using the output module for the format 'MP3'.
'E:/part2.mts' track 12: Using the output module for the format 'MPEG-1/2'.
'E:/part2.mts' track 13: Using the output module for the format 'MP3'.
The file 'C:/myOutput.mkv' has been opened for writing.

The cue entries (the index) are being written...
Muxing took 6 minutes 34 seconds.

warnings:
--- Warnings emitted by job 'merging to file »myOutput.mkv« in directory »C:/«' started on 2015-07-01 14:56:48 ---
Found at least one B frame without second reference in a non closed GOP.
Found at least one B frame without second reference in a non closed GOP.

errors:
--- Errors emitted by job 'merging to file »'merging to file »myOutput.mkv« in directory »C:/«' started on 2015-07-01 14:56:48 ---

**Comment by drandreaskrueger** [comment]
If you need me to look up anything in the source files, I am happy to do that, just tell me what info you need. They are too big to simply upload, I guess. 

Thanks!

**Comment by mbunkus** [comment]
You will have to »append« the second file, not »add« it. Right-click on the file list and you'll see the option for appending files.

Adding = the content of all added files will be laid out in parallel.
Appending = the content of the appended file will start after all of the content of the file it has been appended to.

**Comment by drandreaskrueger** [comment]
Yes, that was it. 

I would have NEVER found that myself, just because drag-n-dropping the files into the input section seemed so intuitive, and the default actions there is "add" not "append".

But: Yiehah. Thanks!

**Comment by mbunkus** [comment]
If you often want to append via drag & drop then you can make the GUI ask you whether you want to add or append the dropped files. Look for »Always add files dropped from external applications« and disable that.

**Comment by drandreaskrueger** [comment]
Very nice. 
Done that now. 
Makes a lot of sense.

BUT: New problem. Can it only connect 2 files?

    More than one track is to be appended to the track 0 
    from file no. 0 ('G:/part_1.m4v'). The argument for '--append-to' was invalid.

Sorry for asking all these dumb questions. I have started reading into the manual https://www.bunkus.org/videotools/mkvtoolnix/doc/mkvmerge.html and tried out this CLI command, because later I want to thread it together with handbrake anyways ...

    mkvmerge.exe -o combined.mkv part_1.m4v +part_2.m4v +part_3.m4v +part_4.m4v +part_5.m4v +part_6.m4v +part_7.m4v

but it won't do it.

**Comment by mbunkus** [comment]
No, it can append more than two files, but unfortunately there's a bug in release 8.1.0 that prevents this from working. It's been fixed already; see #1313. You should get one of the latest [pre-builds](https://www.bunkus.org/videotools/mkvtoolnix/windows/pre/) which contain the fix.

**Comment by mbunkus** [comment]
BTW, that mkvmerge command you've posted should work just fine assuming the track and codec parameters of the tracks match. Note that mkvmerge cannot append tracks produced with different encoder settings (in such a case you'd get an error message stating that the codec parameters don't match).

**Comment by drandreaskrueger** [comment]
yes, the GUI from the pre-build allows to append several files now. Great! Thanks.

... but you are right, that is the next problem now:

    Error: The track number 11 from the file 'chunk_2.ts' cannot be appended 
    to the track number 11 from the file 'chunk_1.ts'. The track parameters do not match.

I tried with the already compressed handbrake outputs 'chunk_x.mkv', then I went back to the original uncompressed 'chunk_x.ts' - but always the same error message: "The track parameters do not match."

What to do now? 

Apart from giving up, or using Adobe Premiere, or Windows Movie Maker - which is fine for one movie, but I want to create a toolchain, to easily store away TV recordings from that SAT-TV-receiver.  

All those 7 chunks were part of the same recording! Why are the parameters different?
Does the SAT-TV-receiver perhaps change recording settings depending on what the TV station is broadcasting in that moment?

Can I ask mkvmerge to print me a table of all the "parameters" of all 7 chunks, so that we can have a look at them?

thx

**Comment by mbunkus** [comment]
I cannot tell you why they're different, only that they're different. mkvmerge is not a video encoder, it cannot magically make such tracks be compatible.

A SAT TV receiver is not doing any encoding/recoding. It only saves onto disk what the TV station is sending out, usually unmodified.

No, mkvmerge cannot print a human-readable list of such parameters because it doesn't fully understand the CodecPrivate formats (it doesn't have to, as it isn't a video decoder). Sorry.

**Comment by drandreaskrueger** [comment]
oh what a pity. 
it would be sad to give up.

Idea: 
Could I perhaps hardcode certain parameter settings into the chunks 
by choosing a definite parameter set in handbrake?

i.e. 

(first step) handbrake, 
recode all (perhaps very differently parametrized) chunks into one set format

(second step) mkvmerge 
of all those 7 parts which should now fit to each other because handbrake made them that way?


?

**Comment by mbunkus** [comment]
I don't know, I've never done that myself. I'm sorry, but I really cannot help you with such a task.

**Comment by drandreaskrueger** [comment]
Thanks anyways. Enjoy your summer!

**Comment by akru13** [comment]
After merge second file goes gray in video player!  [Media Player Classic or VLC]!!!
After merge [first add 1.mkv video then append 2.mk video and click "Mux"], when play merged video the first video is good! but the second video [in merged video] show gray screen and only play audio!
Two videos have exactly same codec.
Video Properties:
General
Unique ID                      : 137528981404392727524386849788205632145 

(0x67771E4C3C978755F71C954EF07E8A91)
Complete name                  : C:\Users\AK\Desktop\Concert\1.mkv
Format                         : Matroska
Format version                 : Version 2
File size                      : 817 MiB
Duration                       : 50mn 1s
Overall bit rate               : 2 283 Kbps
Encoded date                   : UTC 2016-11-11T08:58:32Z
Writing application            : HandBrake 0.10.5 2016021100
Writing library                : Lavf55.12.0 / Lavf55.12.0

Video
ID                             : 1
Format                         : AVC
Format/Info                    : Advanced Video Codec
Format profile                 : Baseline@L3
Format settings, CABAC         : No
Format settings, ReFrames      : 2 frames
Codec ID                       : V_MPEG4/ISO/AVC
Duration                       : 50mn 1s
Bit rate                       : 1 981 Kbps
Width                          : 720 pixels
Height                         : 406 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Constant
Frame rate                     : 29.970 (30000/1001) fps
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Scan type                      : Progressive
Bits/(Pixel*Frame)             : 0.226
Stream size                    : 709 MiB (87%)
Writing library                : x264 core 142 r2479 dd79a61
Encoding settings              : cabac=0 / ref=2 / deblock=1:0:0 / 

analyse=0x1:0x111 / me=hex / subme=6 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / 

me_range=16 / chroma_me=1 / trellis=1 / 8x8dct=0 / cqm=0 / deadzone=21,11 / 

fast_pskip=1 / chroma_qp_offset=-2 / threads=6 / lookahead_threads=1 / 

sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / 

constrained_intra=0 / bframes=0 / weightp=0 / keyint=300 / keyint_min=30 / 

scenecut=40 / intra_refresh=0 / rc_lookahead=30 / rc=crf / mbtree=1 / crf=20.0 / 

qcomp=0.60 / qpmin=0 / qpmax=69 / qpstep=4 / vbv_maxrate=10000 / 

vbv_bufsize=10000 / crf_max=0.0 / nal_hrd=none / filler=0 / ip_ratio=1.40 / 

aq=1:1.00
Default                        : Yes
Forced                         : No
Color range                    : Limited
Color primaries                : BT.601 NTSC
Transfer characteristics       : BT.709
Matrix coefficients            : BT.601

Audio
ID                             : 2
Format                         : MPEG Audio
Format version                 : Version 1
Format profile                 : Layer 3
Mode                           : Joint stereo
Mode extension                 : MS Stereo
Codec ID                       : A_MPEG/L3
Codec ID/Hint                  : MP3
Duration                       : 50mn 1s
Bit rate mode                  : Constant
Bit rate                       : 256 Kbps
Channel(s)                     : 2 channels
Sampling rate                  : 44.1 KHz
Compression mode               : Lossy
Delay relative to video        : -25ms
Stream size                    : 91.6 MiB (11%)
Title                          : Stereo
Writing library                : LAME3.98ھھ
Default                        : Yes
Forced                         : No
